drivers/sensor/ak8975/ak8975.c

```diff
@@ -161,16 +154,23 @@ int ak8975_init(const struct device *dev)
 		return -EINVAL;
 	}
 
-	if (ak8975_read_adjustment_data(drv_data) < 0) {
+	if (ak8975_read_adjustment_data(dev) < 0) {
 		return -EIO;
 	}
 
 	return 0;
 }
 
-struct ak8975_data ak8975_data;
+static struct ak8975_data ak8975_inst_data;
+
+static const struct ak8975_config ak8975_inst_config = {
+	.i2c = I2C_DT_SPEC_INST_GET(0),
+	IF_ENABLED(DT_NODE_HAS_STATUS(DT_INST(0, invensense_mpu9150), okay),
+		   (.mpu9150_i2c = { .bus = DEVICE_DT_GET(DT_BUS(DT_INST(0, invensense_mpu9150))),
+				     .addr = DT_REG_ADDR(DT_INST(0, invensense_mpu9150)) },))
```

Comment:
Can't you use I2C_DT_SPEC_GET to initialize .mpu9150_i2c?

Something like:

`.mpu9150_i2c = I2C_DT_SPEC_GET(DT_BUS(DT_INST(0, invensense_mpu9150))`

Reply:
Not sure how that's generalizable with multi-instancing, what happens for example if instance 1 of ak8975 is connected to instance 0 of mpu9150?